### PR TITLE
Additional load commands

### PR DESCRIPTION
**Pull request by qmfrederik**
- MachEncryptionInfo
- MachRunPath
- MachSourceVersion
- MachUuid

### CODE REVIEW

File: Melanzana.MachO/MachReader.cs [hunk 1]
```diff
@@ -329,11 +359,15 @@ private static MachObjectFile ReadSingle(FatArchHeader? fatArchHeader, MachMagic
...
+                    MachLoadCommandType.DyldInfo => ReadDyldInfo(loadCommandHeader.CommandType, loadCommandPtr, objectFile, stream),
                     MachLoadCommandType.DyldInfoOnly => ReadDyldInfo(loadCommandHeader.CommandType, loadCommandPtr, objectFile, stream),
                     MachLoadCommandType.TowLevelHints => ReadTwoLevelHints(loadCommandPtr, objectFile, stream),
+                    MachLoadCommandType.Uuid => ReadUuid(loadCommandPtr),
+                    MachLoadCommandType.SourceVersion => ReadSourceVersion(loadCommandPtr, objectFile.IsLittleEndian),
+                    MachLoadCommandType.EncryptionInfo64 => ReadEncryptionInfo64(loadCommandPtr, objectFile.IsLittleEndian),
+                    MachLoadCommandType.Rpath => ReadRunPath(loadCommandPtr, objectFile.IsLittleEndian),
                     _ => new MachCustomLoadCommand(loadCommandHeader.CommandType, loadCommandPtr.Slice(LoadCommandHeader.BinarySize, (int)loadCommandHeader.CommandSize - LoadCommandHeader.BinarySize).ToArray()),
-                });
+                }); ;
```
Comment (filipnavara) [inline Melanzana.MachO/MachReader.cs:370]: nit: Unnecessary `;`

File: Melanzana.MachO/MachWriter.cs [hunk 2]
```diff
@@ -301,6 +302,69 @@ private static void WriteDyldInfoCommand(Stream stream, MachLoadCommandType comm
...
+                isLittleEndian);
+
+            Span<byte> encryptionInfoBuffer = stackalloc byte[MachEncryptionInfo.BinarySize];
+            encryptionInfo.Write(encryptionInfoBuffer, isLittleEndian, out var _);
+            stream.Write(encryptionInfoBuffer);
+        }
+
+        private static void WriteRunPath(Stream stream, MachRunPath runPath, bool isLittleEndian, bool is64Bit)
+        {
+            var valueLength = 4 + Encoding.UTF8.GetByteCount(runPath.RunPath) + 1;
```
Comment (filipnavara) [inline Melanzana.MachO/MachWriter.cs:346]: There's couple of null reference type warnings. Is the `RunPath` really supposed to be `string?`?

**Review by filipnavara — COMMENTED** [review]
Few small comments, otherwise LGTM

**Comment by qmfrederik** [comment]
Thanks @filipnavara . I addressed your comments and added a roundtrip test using an executable which contains a LC_RPATH command.  I couldn't easily create one which contained LC_ENCRYPTION_INFO.

Let me know if you have more feedback.

**Review by filipnavara — APPROVED** [review]

**Comment by filipnavara** [comment]
Thanks a lot!

**Pull request merged**